.github/workflows/release.yml

```diff
@@ -71,3 +70,35 @@ jobs:
           draft: false
           files: |
             artifacts/*
+
+  build-vscode-extension:
+    name: Build vscode extension
+    needs: build-binaries
+    runs-on: ubuntu-22.04
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: recursive
+
+      - name: Download binaries artifacts
+        uses: actions/download-artifact@v3
+        with:
+          path: vscode
+
+      - name: Build vscode extension
+        run: |
+          cd vscode
+          chmod +x bin/*
+          npm install
+          npm run vsix
+
+      - name: Upload Release Assets
+        id: upload-release-assets
+        uses: softprops/action-gh-release@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          draft: false
+          files: |
+            vscode/*.vsix
```

Comment:
I really like this. I was surprised that the vsix file containing all platforms is smaller than downloading a single uncompressed executable, but I guess that makes sense.

In the past I created the vsix file for upload to the vscode marketplace offline, but this new build process makes that impossible. Would you mind looking into what it takes to do this upload from CI? I can create needed tokens etc.

Reply:
I added Personal Access Token (PAT) check and publish setp into release workflow.

I did a simple test on the Publish command (it can be built and packaged successfully and request me to provide a token). Because I didn't have a token, I didn't completely test whether Publish is completely correct.